src/installer/destinations.py

```diff
@@ -138,7 +138,7 @@ def __init__(
 
     def _path_with_destdir(self, scheme: Scheme, path: str) -> str:
         file = os.path.join(self.scheme_dict[scheme], path)
-        if self.destdir is not None:
+        if self.destdir:
```

Comment:
I don't think we should handle it like this in the Python side, and instead this should be a CLI-specific thing. I'd much prefer that the CLI pass None to these objects, rather than changing the behaviour of empty strings within the API here.

I'd also be OK with disallowing empty strings in the API, just to avoid this failure mode in downstream code as well.

Reply:
Also, I'd appreciate it if you could add tests for the relevant behaviour here as well. :)